tests/Model/BSONArrayTest.php

```diff
@@ -110,4 +110,15 @@ public function testSetState(): void
         $this->assertInstanceOf(BSONArray::class, $array);
         $this->assertSame($data, $array->getArrayCopy());
     }
+
+    public function testBsonSerializeWithReverseKeys(): void
+    {
+        $array = new BSONArray([]);
+        $array[2] = 'foo';
+        $array[1] = 'bar';
+        $array[0] = 'baz';
+
+        // Since BSONArray uses array_values internally, the reverse order is expected
```

Comment:
Per my earlier comments, I'm not sure I'd refer to this as "reverse order". Rather, I'd say that "keys are disregarded" as the array's natural order is used.